sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -139,8 +139,8 @@ statement
         DROP (IF EXISTS)? partitionSpec (',' partitionSpec)*           #dropTablePartitions
     | ALTER TABLE tableIdentifier partitionSpec? SET locationSpec      #setTableLocation
     | ALTER TABLE tableIdentifier RECOVER PARTITIONS                   #recoverPartitions
-    | DROP TABLE (IF EXISTS)? tableIdentifier PURGE?                   #dropTable
-    | DROP VIEW (IF EXISTS)? tableIdentifier                           #dropTable
+    | DROP TABLE (IF EXISTS)? multipartIdentifier PURGE?               #dropTable
+    | DROP VIEW (IF EXISTS)? multipartIdentifier                       #dropView
```

Comment:
do we have view support in DS v2? This is surprising to me, as I thought we may treat table/view differently in DS v2. Personally I think it's a bad idea that hive unifies table and view, it's weird to see the `sqlText` field in Hive `Table` class.

Reply:
Echo your sentiment towards Hive view handling. Separated visitView from visitTable and added DropViewStatement in preparation for better view handling, possibly even view catalog.

Reply:
Are you going to add `DropView` operator in this PR? If not I think it's better to not add `DropViewStatement` in this PR. We can have a separated PR to add the view support.

I think we haven't reached a consensus that DROP VIEW is implemented via `TableCatalog.dropTable`.

Reply:
I see, separate visitDropView makes it possible to delay moving DropView to catalyst. I am ok either way.

Reply:
I think since these were mixed together originally, it makes sense to move both statements to create parsed plans at the same time. We are going to need to do this eventually anyway.

Why change to use a separate `visitDropView`, but not also move it to Catalyst and use a parsed plan? Is there value in keeping it that way?

Reply:
Yes Hive mixes DROP TABLE and DROP VIEW, but it doesn't mean we have to follow it in data source v2. At least this needs to be discussed.

It seems weird to me that we forward the DROP VIEW request to `TableCatalog` implementations, without official view support in data source v2.

Reply:
@cloud-fan, that's not what this PR does. This PR moves parsing into Catalyst by introducing a parsed plan, `DropViewStatement`, for `DROP VIEW`. We can still convert that to a `DropTable` in the analyzer if we choose to later, like we do for v1. (But I think it makes much more sense to use a `ViewCatalog`.)

The important thing is that we want to keep moving parsing code into Catalyst using parsed plans to separate the implementation (DataSource logical plans) from the parsing logic so that the parser isn't tightly coupled to DataSource v1.

Reply:
ah you are right! sorry I read the PR too fast